libraries/liveIntentId/externalIdSystem.js

```diff
@@ -0,0 +1,173 @@
+import { logError } from '../../src/utils.js';
+import { gdprDataHandler, uspDataHandler, gppDataHandler } from '../../src/adapterManager.js';
+import { submodule } from '../../src/hook.js';
+import { DEFAULT_AJAX_TIMEOUT, MODULE_NAME, parseRequestedAttributes, composeIdObject, eids, GVLID, PRIMARY_IDS } from './shared.js'
+
+// Reference to the client for the liQHub.
+let cachedClientRef
+
+/**
+ * This function is used in tests.
+ */
+export function resetSubmodule() {
+  cachedClientRef = undefined
+}
+
+window.liQHub = window.liQHub ?? []
+
+function initializeClient(configParams) {
+  // Only initialize once.
+  if (cachedClientRef != null) return cachedClientRef
+
+  const clientRef = {}
+
+  const clientDetails = { name: 'prebid', version: '$prebid.version$' }
+
+  const collectConfig = configParams.liCollectConfig ?? {};
+
+  let integration
+  if (collectConfig.appId != null) {
```

Comment:
is it worth checking for the optional `publisherId` param here?

Reply:
@ncolletti The code here determines the type of the integration that depends on a combination of certain fields. In this case if the appId is provided, the actual value of the publisherId can be defined or undefined. Both is fine.